src/App.js

```diff
@@ -15,10 +16,41 @@ function App() {
     });
   }
 
+  function editCard() {
+    setCardState({
+      ...cardState,
+      edited: !cardState.edited,
+      checked: (cardState.checked ? !cardState.checked : cardState.checked)
+    });
+  }
+
+  function captionChange(event) {
+    setCardState({
+      ...cardState,
+      caption: event.target.value,
+    });
+  }
+
+  function textChange(event) {
+    setCardState({
+      ...cardState,
+      text: event.target.value
+    });
+  }
+
+  function cancelEditing() {
+    setCardState({
+      caption: 'Caption', text: 'Text...',
+      edited: !cardState.edited,
+      checked: (cardState.checked ? !cardState.checked : cardState.checked)
+    });
+  }
```

Comment:
You should not set these properties' values hard-coded, but should refer to some storage. As a common practice, you can create a separate state property like 'tempCard' or whatever and place there card data once you're putting updates. card data object itself should not be changed until save icon is pressed, so you could use temp object to apply those changes. Once save icon is pressed - copy data from temp object to original one. Once cancel icon pressed - on reverse, copy data from original object to temporary one,